### PR TITLE
update to opentelemetry 0.24

### LINKED ISSUE #177: Support for opentelemetry 0.23.0
Hi 👋🏽

Would it be possible to add support for opentelemetry 0.23?

### PR DESCRIPTION
**Pull request by mellowagain**
fixes #177

# Checklist

- [x] Changelog updated

### CODE REVIEW

**Comment by mellowagain** [comment]
`tracing-opentelemetry` is only on otel 0.23, theres currently a pr open to update it to 0.24: https://github.com/tokio-rs/tracing-opentelemetry/pull/155

once that one is merged and released we can continue with the otel 0.24 upgrade. im pausing this pr for now

**Comment by mellowagain** [comment]
ran into another blocker which seems not that easily solveable, the `prometheus-exporter` feature depends on `opentelemetry_prometheus`, which is not updated to the new `opentelemetry` version and there doesn't even seem to be a plan to do so. on the readme of the project it states

> The development of prometheus exporter has halt until the Opentelemetry metrics API and SDK reaches 1.0. Current implementation is based on Opentelemetry API and SDK 0.23.

others people also running into this: https://github.com/open-telemetry/opentelemetry-rust/issues/1950

seems like this will be hard block for a while until `opentelemetry` reaches 1.0. im not sure if that is going to happen anytime soon, i thought it was blocked until there was consensus on https://github.com/open-telemetry/opentelemetry-rust/issues/1571 but that may be wrong.

we can optionally choose to fork and update `opentelemetry_prometheus` ourselves but that is additional maintainence burden and wouldn't play well if a user of `autometrics` includes the `opentelemetry_prometheus` themselves, as there would be conflicting versions then.

File: CHANGELOG.md
Comment (hatchan) [inline CHANGELOG.md:12]: Do we have to mark this as a breaking change?

File: examples/opentelemetry-push-custom/Cargo.toml
Comment (hatchan) [inline examples/opentelemetry-push-custom/Cargo.toml:13]: Do we have to use the `.1` here?

**Review by hatchan — APPROVED** [review]

**Pull request merged**
